### PR TITLE
feat(server): Allow admin commands in multi

### PR DESCRIPTION
**Pull request by ashotland**
Needed for sentinel support (#706)

Signed-off-by: ashotland <ari@dragonflydb.io>

### CODE REVIEW

File: src/server/conn_context.cc [hunk 1]
```diff
@@ -15,6 +16,20 @@ namespace dfly {
 
 using namespace std;
 
+StoredCmd::StoredCmd(const CommandId* d, CmdArgList args) : descr(d) {
```
Comment (ashotland) [inline src/server/conn_context.cc:19]: Non functional change - moved this logic from main service so all storedcmd args manipulation is in storedcmd

File: src/server/main_service.cc [hunk 2]
```diff
@@ -597,11 +597,6 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
   }
 
   if (under_multi) {
-    if (cid->opt_mask() & CO::ADMIN) {
```
Comment (ashotland) [inline src/server/main_service.cc:600]: Functional change 1

File: src/server/transaction.cc [hunk 3]
```diff
@@ -335,7 +335,7 @@ bool Transaction::RunInShard(EngineShard* shard) {
 
   // We make sure that we lock exactly once for each (multi-hop) transaction inside
   // transactions that lock incrementally.
-  if (incremental_lock && ((sd.local_mask & KEYLOCK_ACQUIRED) == 0)) {
+  if (!IsGlobal() && incremental_lock && ((sd.local_mask & KEYLOCK_ACQUIRED) == 0)) {
```
Comment (ashotland) [inline src/server/transaction.cc:338]: Functional change 2

File: src/server/conn_context.h [hunk 4]
```diff
@@ -13,13 +13,24 @@
...
 class EngineShardSet;
+class ConnectionContext;
 
 struct StoredCmd {
   const CommandId* descr;
-  std::vector<std::string> cmd;
 
-  StoredCmd(const CommandId* d = nullptr) : descr(d) {
+ private:
+  std::vector<std::string> stored_args;
```
Comment (romange) [inline src/server/conn_context.h:22]: underscore at the end of private members.

File: src/server/conn_context.h [hunk 5]
```diff
@@ -13,13 +13,24 @@
...
+  CmdArgVec arg_vec;
+  CmdArgList arg_list;
+
+ public:
+  StoredCmd(const CommandId* d, CmdArgList args);
+
+  CmdArgList& ArgList() {
```
Comment (romange) [inline src/server/conn_context.h:29]: no need to return a reference, it's a span, shallow copy. Return by value and make the function const

File: src/server/transaction.cc (hunk 3 above)
Comment (romange) [reply src/server/transaction.cc:338]: I do not remember by heart the logic here, unfortunately. Can you explain the reason for this change?

**Comment by ashotland** [comment]
@romange FYI all done

**Review by romange — APPROVED** [review]

**Pull request merged**